File: src/game/tools/rpc/output/wrap/IntKVWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type IntKVWrap struct {
	
	Key int16 // map的key
	Value int16 // map的value
	rpc.Wrapper
}

func (w *IntKVWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.Key = pck.PopInt16(); 
	w.Value = pck.PopInt16(); 
	return w
}

func (w *IntKVWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt16(w.Key) 
	pck.PutInt16(w.Value) 
}

File: src/game/tools/rpc/output/wrap/IntKVWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class IntKVWrap : IWrapper
	{	
		public short Key; // map的key
		public short Value; // map的value
		
		public void Decode(Packet pck)
		{	
			Key = pck.GetShort(); 
			Value = pck.GetShort(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutShort(Key); 
        	pck.PutShort(Value); 
        }
	}
}